test/integration/mecanum_drive_system.cc

```diff
@@ -0,0 +1,537 @@
+      test::Relay velocityRamp;
+      const int movementDirection = (forward ? 1 : -1);
+      double desiredLinVelX = movementDirection * 0.15;
+      double desiredLinVelY = movementDirection * 0.15;
+      velocityRamp.OnPreUpdate(
+          [&](const UpdateInfo &/*_info*/,
+              const EntityComponentManager &)
+          {
+            msgs::Set(msg.mutable_linear(),
+                      math::Vector3d(desiredLinVelX, desiredLinVelY, 0));
+            pub.Publish(msg);
+          });
+
+      server.AddSystem(velocityRamp.systemPtr);
+
+      server.Run(true, 3000, false);
+
+      // Poses for 4s
+      ASSERT_EQ(4000u, poses.size());
+
+      int sleep = 0;
+      int maxSleep = 30;
+      for (; odomPoses.size() < 150 && sleep < maxSleep; ++sleep)
+      {
+        std::this_thread::sleep_for(std::chrono::milliseconds(100));
+      }
+      ASSERT_NE(maxSleep, sleep);
+
+      // Odometry calculates the pose of a point that is located half way
+      // between the two wheels, not the origin of the model. For example,
+      // if the vehicle is commanded to rotate in place, the vehicle will
+      // rotate about the point half way between the two wheels, thus,
+      // the odometry position will remain zero.
+      // However, since the model origin is offset, the model position will
+      // change. To find the final pose of the model, we have to do the
+      // following similarity transformation
+      
+      math::Pose3d tOdomModel(0.554283,0,-0.325,0,0,0);
+      auto finalModelFramePose =
+          tOdomModel * odomPoses.back() * tOdomModel.Inverse();
+
+      // Odom for 3s
+      ASSERT_FALSE(odomPoses.empty());
+      EXPECT_EQ(150u, odomPoses.size());
+
+      auto expectedLowerPosition =
+          (forward ? poses[0].Pos() : poses[3999].Pos());
+      auto expectedGreaterPosition =
+          (forward ? poses[3999].Pos() : poses[0].Pos());
+
+      EXPECT_LT(expectedLowerPosition.X(), expectedGreaterPosition.X());
+      EXPECT_LT(expectedLowerPosition.Y(), expectedGreaterPosition.Y());
+      EXPECT_NEAR(expectedLowerPosition.Z(), expectedGreaterPosition.Z(), tol);
+      EXPECT_NEAR(poses[0].Rot().X(), poses[3999].Rot().X(), tol);
+      EXPECT_NEAR(poses[0].Rot().Y(), poses[3999].Rot().Y(), tol);
+      EXPECT_NEAR(poses[0].Rot().Z(), poses[3999].Rot().Z(), tol);
+
+      // The value from odometry will be close, but not exactly the ground truth
+      // pose of the robot model. This is partially due to throttling the
+      // odometry publisher, partially due to a frame difference between the
+      // odom frame and the model frame, and partially due to wheel slip.
+      EXPECT_NEAR(poses.back().Pos().X(), odomPoses.back().Pos().X(), 1e-2);
+      EXPECT_NEAR(poses.back().Pos().Y(), odomPoses.back().Pos().Y(), 1e-2);
+      // EXPECT_NEAR(poses.back().Pos().X(), finalModelFramePose.Pos().X(), 1e-2);
+      // EXPECT_NEAR(poses.back().Pos().Y(), finalModelFramePose.Pos().Y(), 1e-2);
```

Comment:
Is there a reason for commenting these out? `finalModelFramePose` is now an unused variable.

Reply:
i missed them, i put them back in and tested them